test/pytest/service-desktop/test_outbox.py

```diff
@@ -0,0 +1,183 @@
+        except TransactionError:
+            return False
+        else:
+            return True
+
+    def add_message(self, message_number, message_body):
+        try:
+            message = AddMessage(message_number, message_body).run(self.harness).message
+        except TransactionError:
+            return False
+        else:
+            return True, message
+
+    def delete_message_by_id(self, message_record_id):
+        try:
+            DeleteMessageById(message_record_id).run(self.harness)
+        except TransactionError:
+            return False
+        else:
+            return True
+
+    def add_contact(self, contact_record):
+        try:
+            contact_id = AddContact(contact_record).run(self.harness).id
+        except TransactionError:
+            return False
+        else:
+            return True, contact_id
+
+    def delete_contact_by_id(self, contact_record_id):
+        try:
+            DeleteContactById(contact_record_id).run(self.harness)
+        except TransactionError:
+            return False
+        else:
+            return True
+
+
+@pytest.mark.service_desktop_test
+@pytest.mark.usefixtures("phone_unlocked")
+def test_getting_notifications(harness):
+    outbox_tester = OutboxTester(harness)
+    result, received_notifications = outbox_tester.get_notifications()
+    assert result, "Failed to get notifications!"
+    assert not received_notifications, "Notification list is not empty at the beginning of the test!"
+
+    # Add message to generate "message created" and "thread created" notifications
+    message_number = "123456789"
+    message_body = "Hello, how are you?"
+
+    result, message_record = outbox_tester.add_message(message_number, message_body)
+    assert result, "Failed to add message!"
+
+    # Add contact to generate "contact created" notification
+    contact_record = {
+        "address": "6 Czeczota St.\n02600 Warsaw",
+        "altName": "Smith",
+        "email": "john.smith@mudita.com",
+        "blocked": False,
+        "favourite": False,
+        "ice": False,
+        "numbers": [
+            "123456789"
+        ],
+        "speedDial": "1",
+        "priName": "John",
+        "note": "Some note"
+    }
+
+    result, contact_id = outbox_tester.add_contact(contact_record)
+    assert result, "Failed to add contact!"
+
+    result, received_notifications = outbox_tester.get_notifications()
+    assert result, "Failed to get notifications!"
+
+    # Check if all notifications are present
+    found_message_notification = False
+    for notification in received_notifications:
+        if notification["type"] == 1 and notification["change"] == 1\
+                and notification["record_id"] == message_record["messageID"]:
+            found_message_notification = True
+            break
+    found_thread_notification = False
+    for notification in received_notifications:
+        if notification["type"] == 2 and notification["change"] == 1:
```

Comment:
why not add the change into the parser? same for type :) this is partially what endpoint parsers are made: https://github.com/mudita/QAMuditaHarness/pull/37/files#r806788667 <- related comment

Reply:
Done